utils.go

```diff
@@ -30,6 +31,24 @@ func generateInstanceId() string {
 	return prefix
 }
 
+func getConnectionId() string {
```

Comment:
can we link to where we got this from. It's not a typical code someone writes every day

Reply:
It is a new implementation. It's similar to
https://github.com/google/uuid/blob/2d3c2a9cc518326daf99a383f07c4d3c44317e4d/version4.go#L47-L56 and https://github.com/hprose/hprose-go/blob/83de97da5004027694d321ca38c80fca3fac98c2/uuid.go#L91-L98
(because it does the same thing).